### PR TITLE
Comments, commentary, stuff.

### PR DESCRIPTION
**Pull request by paytonrules**
- Some feedback, mostly small stuff.

This is pretty darn good. The general design is nice, you've got things arranged similarly to how I might do it. The only thing I see is making sure you've got those three rules practiced for TDD. There's a lot of tiny corner cases missed.

### CODE REVIEW

File: httpserver/test/httpserver/socket_test.clj [hunk 1]
```diff
@@ -4,23 +4,23 @@
...
 
 (deftest test-open
   (let [server-socket (open 5000)]
     (testing "Create a ServerSocket"
       (is (instance? ServerSocket server-socket)))
-    (testing "Socket listening on given port"
+    (testing "Socket listening on given port" ; do you know it's really listening?
       (is (= 5000 (.getLocalPort server-socket))))
-    (.close server-socket)))
+    (.close server-socket))) ; what happens if there is an error above? Won't the socket stay open
```
Comment (hnlee) [inline httpserver/test/httpserver/socket_test.clj:16]: @paytonrules This is the comment about which I said in Slack:

> but I think if either of the two assertions in the test are wrong, the ServerSocket would either not exist or that line closing the ServerSocket would itself return an error? I don’t think it would keep it open
